### PR TITLE
db/hints: Use dedicated functions to lock a shared mutex

### PR DESCRIPTION
**Pull request by dawmd**
Seastar has functions implementing locking a `seastar::shared_mutex`. We should use those now instead of reimplementing them in Scylla.

### CODE REVIEW

File: db/hints/manager.cc [hunk 1]
```diff
@@ -286,18 +287,8 @@ sync_point::shard_rps manager::calculate_current_sync_point(std::span<const gms:
...
-        co_await _migration_mutex.lock_shared();
-
-        try {
-            co_return lock_type{_migration_mutex, std::adopt_lock_t{}};
-        } catch (...) {
-            _migration_mutex.unlock_shared();
-            throw;
-        }
-   }));
+    const auto shared_lock = co_await get_shared_lock(_migration_mutex);
```
Comment (piodul) [inline db/hints/manager.cc:291]: Now that we use `get_{shared,unique}_lock` and don't name the type of the lock I think we can also drop the `<shared_mutex>` import, right?

Comment (dawmd) [reply db/hints/manager.cc:291]: Yes, that's a good point

**Comment by dawmd** [comment]
Version 2:
* ditched `#include <shared_mutex>`

**Review by piodul — APPROVED** [review]

**Comment by piodul** [comment]
@dawmd If you send a single commit PR, please make sure that the commit message is the same as the PR description. The commit you sent is missing the description. The script that maintainers use for queueing PRs just cherry-picks the commit in this case and completely skips creating a merge commit which would usually contain the PR description.

I fixed the commit message for you, but from now on please remember to keep them in sync. Next time, a maintainer might miss it and queue a commit without a proper description.

**Pull request closed without merging**